### PR TITLE
fix: BeaconBeamFeature not rendering correctly

### PR DESCRIPTION
**Pull request by kristofbolyai**

### CODE REVIEW

**Comment by Incompleteusern** [comment]
How does this fix the issue?

File: common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java [hunk 1]
```diff
@@ -32,14 +33,16 @@ public List<Class<? extends Model>> getModelDependencies() {
...
+        if (compassLocation.isEmpty()) return;
 
         PoseStack poseStack = event.getPoseStack();
         MultiBufferSource.BufferSource bufferSource =
                 McUtils.mc().renderBuffers().bufferSource();
 
         Vec3 camera = event.getCamera().getPosition();
-        Location location = CompassModel.getCompassLocation().get();
+        Location location = compassLocation.get();
+        location.set(location.x, 0, location.z);
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java]: Will this modify the underlying representation, or do we return a copy?

**Comment by kristofbolyai** [comment]
> How does this fix the issue?

y is -1 otherwise, setting to 0 actually renders it.

File: common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java]: It modifies the representation. Why?

**Comment by Incompleteusern** [comment]
> > How does this fix the issue?
> 
> y is -1 otherwise, setting to 0 actually renders it.

Why is y -1?

**Comment by kristofbolyai** [comment]
> > > How does this fix the issue?
> > 
> > 
> > y is -1 otherwise, setting to 0 actually renders it.
> 
> Why is y -1?

Because some POIs only have a x and z coordinate.

**Comment by Incompleteusern** [comment]
> 

Which POIs? and shouldn't we be applying a fix somewhere else.

**Comment by Incompleteusern** [comment]
```java
        if (locationSupplier != null && locationSupplier.get() != null) {
            WaypointPoi waypointPoi = new WaypointPoi(() -> {
                Location location = locationSupplier.get();

                // Make sure to always render on top
                location.set(location.x, Double.MAX_VALUE, location.z);

                return MapLocation.fromLocation(location);
            });

            return Optional.of(waypointPoi);
        }
```

I think this is responsible as it modifies the underlying representation, also the setting to max double is likely unnecessary now.

File: common/src/main/java/com/wynntils/wynn/model/CompassModel.java [hunk 2]
```diff
@@ -48,9 +48,6 @@ public static Optional<WaypointPoi> getCompassWaypoint() {
             WaypointPoi waypointPoi = new WaypointPoi(() -> {
                 Location location = locationSupplier.get();
 
-                // Make sure to always render on top
-                location.set(location.x, Double.MAX_VALUE, location.z);
-
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/CompassModel.java:53]: This was the problem, we did not even need this as rendering code already handled it. @Incompleteusern

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/wynn/model/CompassModel.java:53]: Thanks!

**Review by Incompleteusern — APPROVED** [review]

File: common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java]: I'm not entirely happy about the compass location API. It feels too open and leaky. I realize there's a difference between locations that are fixed (like NPCs) and that can change (like compass point), but the API should be more restrictive here, I think. I noticed you and Incomplete did some stuff on static vs dynamic poi's that I've not had the time to look into detail. Maybe it's the beginning on getting this more strict.

Also, we should perhaps consider having a location representation where we can actually flag the Y value as "don't know/don't care".

File: common/src/main/java/com/wynntils/wynn/model/CompassModel.java (hunk 2 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/wynn/model/CompassModel.java:53]: I've been saying for a long time that this was fishy code. ;-)

**Review by magicus — APPROVED** [review]

**Pull request merged**